std/hash/hash.go

```diff
@@ -30,3 +30,5 @@ type Hash interface {
 	// Reset empty the internal state and put the intermediate state to zero.
 	Reset()
 }
+
+var BuilderRegistry = make(map[string]func(api frontend.API) (Hash, error))
```

Comment:
what is that? comment?

Reply:
That's for the GKR API. In the specification of a circuit, we don't pass an actual hash object, but a string identifier which can later be used to construct a hash object both in the SNARK (for the GKR verifier) and in pure Go (for the GKR prover.) So this would be used to "register" hash functions in a similar way to hints.